More descriptive API error messages

pastDexter opened the pull request:
Currently we are not checking API response status and straight converting it to `json`, so if API returned some text error, we are exposing it to users as `JOSN.parse` errors and not original errors from API

Related to #52

Bombasarkadian commented:
![Screenshot 2020-03-31 at 11 44 39](https://user-images.githubusercontent.com/2972266/78012331-15914100-7345-11ea-9fdc-6436a7e1620e.png)
I got this error when trying to join a private room

pastDexter commented:
@Bombasarkadian I can't reproduce. Can you check?

1. I added proper error catching for this case, because it wasn't handled before.
2. To somehow solve this race condition when leaving/joining room I added retry mechanism to the join function in `GamePage`
